examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+                    }
+                }
+                catch (FormatException)
+                {
+                    // If its not in valid base64 format, ignore the message 
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static void BackupKeyValues(HashSet<KeyLabel> updatedKeyLabels, ILogger log)
+        {
+            // Connect to Azure App Configuration stores
+            ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+            ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+
+            // Read all settings from Primary Store and update secondary store
+            foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettings(new SettingSelector()))
+            {
+                KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                if(updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                {
+                    // current setting retrieved from primary store needs to be updated in secondary store
+                    secondaryAppConfigClient.SetConfigurationSetting(setting);
+                    log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+                    updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                }
+            }
+
+            // Delete key labels still present in updatedKeyLabels from secondary store
+            foreach (var keyLabel in updatedKeyLabels)
+            {
+                secondaryAppConfigClient.DeleteConfigurationSetting(keyLabel.Key, keyLabel.Label);
+                log.LogInformation($"Successfully deleted key: {keyLabel.Key} label: {keyLabel.Label} from secondary store.");
+            }
+        }
+
+        [FunctionName("BackupFromStorageQueue")]
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+                
+                // Create storage queue client
+                QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+
+                // Peek to see if there are events in the queue
+                while (queueClient.PeekMessages().Value.Length > 0 )
+                {
+                    Response<QueueMessage[]> messages = queueClient.ReceiveMessages(maxMessages: 32);
```

Comment:
Can we use `ReceiveMessagesAsync` instead?

Please define 32 as const.

Reply:
done